### PR TITLE
Closes #113 make template adnv

### LINKED ISSUE #113: Template: ADNV
### Feature Idea

Background Information
Showcase from start to finish how to create an ADNV related to UPSIT. This should include:

Parameter mappings​
Endpoint programming​
New computations​ using compute_upsit_percentile() to create age and sex adjusted upsit raw score
Should be based on NV

Definition of Done

New file inst/templates/adnv.R showcasing how to create ADNV

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by jwang-lilly**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: inst/templates/ad_adnv.R [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    AVAL = NVSTRESN,
+    AVALC = if_else(
+      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
+      NVSTRESC,
+      NA
+    )
+  )
+
+## upsit percentile derivations ----
+### Convert UPSITTS to  UPSITPC----
```
Comment (Fanny-Gautier) [inline inst/templates/ad_adnv.R]:
```suggestion
### Convert UPSITTS to UPSITPC----
```

File: inst/templates/ad_adnv.R [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+library(admiral)
+library(admiralneuro)
+# for development purposes remotes::install_github("pharmaverse/pharmaversesdtm")
+library(pharmaversesdtm) # Contains example datasets from the CDISC pilot project
+library(dplyr)
+library(lubridate)
+library(stringr)
+
+# Define project options/variables ----
```
Comment (Fanny-Gautier) [inline inst/templates/ad_adnv.R]: I think we don’t use it in templates, but I couldn’t find the comment with the explanation again.
@manciniedoardo, could you please confirm/clarify?

File: inst/templates/ad_adnv.R [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  filter(NVTESTCD == "UPSIT") %>% # Filter nv dataset for upsit records only
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = NVDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+adnv <- adnv %>%
+  ## Add PARAMCD and PARAM ----
```
Comment (Fanny-Gautier) [inline inst/templates/ad_adnv.R]: `PARAMN` is available within the lookup table. Comment the reason why not to merge it here, or remove it from the lookup table.

File: inst/templates/ad_adnv.R [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+## Derive baseline information ----
+
+### Calculate BASE ----
+adnv <- adnv %>%
+  derive_var_base(
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD, BASETYPE)),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+  ### Calculate BASEC ----
```
Comment (Fanny-Gautier) [inline inst/templates/ad_adnv.R]: Since `AVALC` is always missing, `BASEC` is always missing and could be dropped.

File: inst/templates/ad_adnv.R [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM),
+    by_vars = exprs(NVTESTCD, NVCAT)
+  ) %>%
+  ## Calculate AVAL and AVALC ----
+  # AVALC should only be mapped if it contains non-redundant information.
+  mutate(
+    AVAL = NVSTRESN,
+    AVALC = if_else(
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply inst/templates/ad_adnv.R]: # AVALC should only be mapped if it contains non-redundant information. The if condition is not met.
 
 `    AVALC = if_else(
      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
      NVSTRESC,
      NA
    )`

File: inst/templates/ad_adnv.R [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+## upsit percentile derivations ----
+### Convert UPSITTS to  UPSITPC----
+keep_vars <- c(
+  get_admiral_option("subject_keys"),
+  adsl_vars,
+  exprs(ADT, ADY, VISIT)
+)
+
+adnv <- adnv %>%
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply inst/templates/ad_adnv.R]: populated.

File: inst/templates/ad_adnv.R [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(AGE, SEX, TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+adnv <- nv %>%
+  ## Join ADSL with NV (need TRTSDT for ADY derivation) ----
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply inst/templates/ad_adnv.R]: @Fanny-Gautier, what is the "it" you referred to?

adtpet has the same code.
`
adtpet <- nv %>%
  derive_vars_merged(
    dataset_add = adsl,
    new_vars = adsl_vars,
    by_vars = get_admiral_option("subject_keys")
  ) 
`

template of admiralmetabolic ad_advs.R has the same code as well.

`
advs <- advs %>%
  derive_vars_merged(
    dataset_add = adsl,
    new_vars = adsl_vars,
    by_vars = get_admiral_option("subject_keys")
  )
`

[the remark this answers is not in the document]
Comment (Fanny-Gautier) [reply inst/templates/ad_adnv.R]: Ok then let's go for it. I checked in the core `{admiral}` package and could not find it. I remember a discussion about it where we should avoid using it, but I could not remember whether this was within vignettes or templates.
As available in other extension packages, then let's apply it here.

[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/templates/ad_adnv.R]: sorry, i missed this. happy to go with your approach
